File: sic-api/Entities/Db/DbSubDistrict.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using sic_api.Entities;

namespace sic_api.Entities.Db;

[Index(nameof(SubDistrictCode), IsUnique = true)]
[Table("db_sub_district")]
public class DbSubDistrict : BaseEntity
{
    
    [Required]
    [ForeignKey(nameof(District))]
    [Column("district_id")]
    public Guid DistrictId { get; set; }
    public DbDistrict District { get; set; } = default!;

    [Required]
    [MaxLength(10)]
    [Column("sub_district_code")]
    public string SubDistrictCode { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("sub_district_name_en")]
    public string SubDistrictNameEn { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("sub_district_name_local")]
    public string SubDistrictNameLocal { get; set; } = default!;

    [Required]
    [MaxLength(20)]
    [Column("zip_code")]
    public string ZipCode { get; set; } = default!;

    [Column("latitude")]
    public long? Latitude { get; set; }
    
    [Column("longitude")]
    public long? Longitude { get; set; }

    [Required]
    [Column("is_active")]
    public bool IsActive { get; set; } = false;
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/db/DbSubDistrict.java
package com.softinter.sicapi.entity.db;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "db_sub_district",
       indexes = {
           @Index(name = "idx_sub_district_code", columnList = "sub_district_code", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class DbSubDistrict extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "district_id", nullable = false)
    private DbDistrict district;

    @Column(name = "sub_district_code", nullable = false, length = 10)
    private String subDistrictCode;

    @Column(name = "sub_district_name_en", nullable = false, length = 255)
    private String subDistrictNameEn;

    @Column(name = "sub_district_name_local", nullable = false, length = 255)
    private String subDistrictNameLocal;

    @Column(name = "zip_code", nullable = false, length = 20)
    private String zipCode;

    @Column(name = "latitude")
    private Long latitude;

    @Column(name = "longitude")
    private Long longitude;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;
}
